### ISSUE TITLE
soc training loss is always nan

### ISSUE THREAD

**Issue by upperblacksmith** [ZHKKKe/MODNet#95]
it seems that somebody encount this problem,and the image has normalize 0~1. I have attached the dataloader of soc code as follows,and any advice and help will great appreciate. thanks
![123567](https://user-images.githubusercontent.com/34127965/115665293-ac2c0800-a375-11eb-8c50-0101ba5a2d28.JPG)

**Comment by upperblacksmith** [comment]
the above code has any problem?

**Comment by luoww1992** [comment]
i have the same code as yours, it is ok .

**Comment by upperblacksmith** [comment]
> i have the same code as yours, it is ok .
would you mind tell me the image shape in your training datasets

**Comment by Hvorost** [comment]
@upperblacksmith 
Hi, did you fixed the problem where loss is always nan? I have the same problem.

**Comment by JanJan1995** [comment]
@upperblacksmith
Hey, I also meet this question. In tensorboard, my loss suddenly shows Nan value, I don't know what happened there. You guys can observe the loss change process in my soc_training_iter. The zigzag part of loss curve means error of Nan. Need help, thanks a lot.

![My_dataloader](https://user-images.githubusercontent.com/84489083/120147419-355a1880-c219-11eb-8c6d-c1c7495c210e.png)
![soc_detail_loss](https://user-images.githubusercontent.com/84489083/120147425-368b4580-c219-11eb-88ee-afebc6c625fc.png)
![soc_semantic_loss](https://user-images.githubusercontent.com/84489083/120147428-3723dc00-c219-11eb-8827-457f436bd9fa.png)

**Comment by bratyslav** [comment]
During fine-tuning, some images start to "fade", and as a result predictions tend to zero. The error between almost zero and almost zero is practically zero. As a result, at some point, the error becomes less than the minimum value of float. I have no idea hove to handle it :(

Technically, the neural network really minimizes the error :) But not at all as we would like. 

(First screenshot contains outputs before fine-tuning, second - during the process)
![Screenshot 2021-07-16 at 12 52 21](https://user-images.githubusercontent.com/39599162/125931271-954bb323-be4e-4e76-99cd-38de14e730e2.png)
![Screenshot 2021-07-16 at 12 53 58](https://user-images.githubusercontent.com/39599162/125931281-fa04d3fa-547f-4fc6-a92f-49c77c5de786.png)
